chromium_src/components/sync_device_info/device_info.cc

```diff
@@ -0,0 +1,44 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "components/sync_device_info/device_info.h"
+
+#include <algorithm>
+
+#define ToValue ToValue_ChromiumImpl
+#include "../../../../components/sync_device_info/device_info.cc"
+#undef ToValue
+
+namespace syncer {
+  std::string GetMobileOSString(const std::string user_agent) {
+    if (user_agent.find("IOS") != std::string::npos) {
+	  return "ios";
+	}
+
+	if (user_agent.find("android") != std::string::npos) {
+	  return "android";
+	}
+	return "unknown";
+  }
+
+  std::unique_ptr<base::DictionaryValue> DeviceInfo::ToValue() const {
```

Comment:
we can call `ToValue_ChromiumImpl()` to get the value and then set our own os string additionally.
So we don't have to duplicate logic